model_compression_toolkit/graph_builder/common/base_graph_builder.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2025 Sony Semiconductor Israel, Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+from abc import ABC, abstractmethod
+from typing import Any, Callable, Optional
+
+from model_compression_toolkit.core.common.visualization.tensorboard_writer import TensorboardWriter
+
+from model_compression_toolkit.target_platform_capabilities.targetplatform2framework.framework_quantization_capabilities import \
+    FrameworkQuantizationCapabilities
+
+from model_compression_toolkit.core.common import Graph
+
+
+class BaseGraphBuilder(ABC):
+    """
+    Abstract base class for converting framework-specific models into a framework-agnostic computational graph,
+    and applying standard graph transformations.
+
+    Subclasses must implement methods to:
+    1. Convert a model to a graph representation.
+    2. Apply optional graph simplifications and preprocessing.
+    """
+
+    def build_graph(self,
+                    model: Any,
+                    representative_dataset: Callable = None,
+                    fqc: FrameworkQuantizationCapabilities = None,
+                    tensorboard_writer: TensorboardWriter = None,
+                    linear_collapsing: bool = True,
```

Comment:
what about the substitutions in `get_substitutions_post_statistics_collection`? aren't they graph preparation related? if so, need to think if they need to be part of this module or somewhere else

Reply:
They depend on the quantization params (this is why they are used today only after loading the fqc, and computing quantization params). So I do not think they should be part of it...

Reply:
Let's talk about this offline, I want us to take another look at this an figure out where they should land